driven/storage/adapter_scores.go

```diff
@@ -445,3 +445,47 @@ func (a *Adapter) findRankForScore(score float64, orgID string, appID string) (u
 
 	return uint32(count) + 1, nil
 }
+
+// FindScoresWithoutExternalUserID finds scores that don't have external_user_id populated
+func (a *Adapter) FindScoresWithoutExternalUserID(orgID string, appID string, limit *int, offset *int) ([]model.Score, error) {
+	filter := bson.M{
+		"org_id": orgID,
+		"app_id": appID,
+		"$or": []bson.M{
+			{"external_user_id": bson.M{"$exists": false}},
+			{"external_user_id": ""},
```

Comment:
I think we actually do want to exclude empty string `external_user_id`s from the results to ensure the migration does not get stuck in an infinite loop trying to load IDs for users that cannot be found.